README.md

```diff
@@ -1 +1,200 @@
+```
+
+#### 4. Verify the Log Sink
+
+Replace the "xxxxx" with the appropriate out vertex pod name:
+```bash
+kubectl logs nats-source-e2e-out-0-xxxxx
+```
+
+You should see the output similar to:
+```
+2023/09/05 19:18:44 (out)  Payload -  Hello World  Keys -  []  EventTime -  1693941455870
+```
+
+#### 5. Clean up
+
+To delete the Numaflow pipeline and the Nats server, run:
+```bash
+kubectl delete -k ./example
+```
+
+Congratulations! You have successfully run a Nats source in a Numaflow pipeline on your local kube cluster.
+
+## How to use the Nats source in our own Numaflow pipeline
+
+To integrate the Nats source in your own Numaflow pipeline, follow these detailed steps:
+
+### 1. Deploy your Nats server
+Deploy our own Nats server to our cluster. Refer to the [NATS Docs](https://docs.nats.io/running-a-nats-service/introduction) for guidance.
+
+### 2: Create a ConfigMap
+Define the Nats source configuration in a ConfigMap and mount it to the Nats source pod as a volume. Create a ConfigMap using the example below:
+
+```yaml
+apiVersion: v1
+data:
+  nats-config.yaml: |
+    url: nats
+    subject: test-subject
+    queue: my-queue
+    auth:
+      token:
+        localobjectreference:
+          name: nats-auth-fake-token
+        key: fake-token
+kind: ConfigMap
+metadata:
+  name: nats-config-map
+```
+
+The configuration contains the following fields:
+* `url`: The Nats server URL.
+* `subject`: The Nats subject to subscribe to.
+* `queue`: The Nats queue group name.
+* `auth`: The Nats authentication information.
+  * `token`: The Nats authentication token information.
+    * `name`: The name of the secret that contains the authentication token.
+    * `key`: The key of the authentication token in the secret.
+
+### 3. Specify the Nats Source in the Pipeline
+Name your Nats Configuration ConfigMap as `nats-config.yaml` and mount it to the Nats source pod as a volume under path `/etc/config`.
+Create all the secrets that are referenced in the Nats source configuration and mount them to the Nats source pod as volumes under path `/etc/secrets/{secret-name}`.
+
+Include the Nats Source in your pipeline using the template below:
+
+```yaml
+apiVersion: numaflow.numaproj.io/v1alpha1
+kind: Pipeline
+metadata:
+  name: nats-source-e2e
+spec:
+  vertices:
+    - name: in
+      scale:
+        min: 2
+      volumes:
+        - name: my-config-mount
+          configMap:
+            name: nats-config-map
+        - name: my-secret-mount
+          secret:
+            secretName: nats-auth-fake-token
+      source:
+        udsource:
+          container:
+            image: quay.io/numaio/numaflow-source/nats-source:v0.5.2
+            volumeMounts:
+              - name: my-config-mount
+                mountPath: /etc/config
+              - name: my-secret-mount
+                mountPath: /etc/secrets/nats-auth-fake-token
+    - name: out
+      sink:
+        log: {}
+  edges:
+    - from: in
+      to: out
+```
```

Comment:
just a note that if you do change the image name to change it here too

Reply:
also, wonder if they should run "latest" tag instead of calling out "0.5.2" in particular?

Reply:
I was referring to [numaflow-go](https://github.com/numaproj/numaflow-go/pull/77/files) where we have all our UDFs tagged with "v0.5.0". I think we plan to have semantic versions for NATS source hence I am putting the current "latest" version here. In practice, user can choose to specify which ever version that's compatible with their numaflow version.

Reply:
okay